mutable_tree.go

```diff
@@ -566,13 +566,10 @@ func (tree *MutableTree) deleteVersion(version int64) error {
 	if version <= 0 {
 		return errors.New("version must be greater than 0")
 	}
-	if version == tree.version {
-		return errors.Errorf("cannot delete latest saved version (%d)", version)
-	}
 	if !tree.VersionExists(version) {
 		return errors.Wrap(ErrVersionDoesNotExist, "")
 	}
-	if err := tree.ndb.DeleteVersion(version, true); err != nil {
+	if err := tree.ndb.DeleteVersion(version, false); err != nil {
```

Comment:
Are you sure this works? You have to move every node in the merkle tree for the historical state at the prior height, to being in the live state section of node db

Reply:
I'm not sure, only did basic unit test in cosmos-sdk side here:https://github.com/cosmos/cosmos-sdk/pull/11361, I need more understanding on internals of iavl I think. 
Do you think there is an efficient way to achieve this rollback feature?

Reply:
I think we don't need to patch iavl at all, there's `LoadVersionForOverwriting` which is perfect for rollback state.

Reply:
We should also check if it will be pruned correctly.

Reply:
https://github.com/cosmos/cosmos-sdk/pull/11361/files#diff-54fe31c47e1d9d86bbaf2face4c464c1f5a74e3c81ca5ac9a27a25a157752592R56
it's checked in this unit test, that one can rollback and recreate new states.